modules/data.land/R/write_ic.R

```diff
@@ -16,41 +32,51 @@ write_ic <- function(in.path, in.name, start_date, end_date,
   # Match PFTs
   
   obs <- as.data.frame(veg_info[[2]], stringsAsFactors = FALSE)
-  
-  # NOTE : match_pft may return NAs for unmatched dead trees
-  pft.info <- PEcAn.data.land::match_pft(obs$bety_species_id, pfts)
-  
-  ### merge with other stuff
+
+  # # NOTE : match_pft may return NAs for unmatched dead trees
+  pft.info <- PEcAn.data.land::match_pft(bety_species_id = obs$bety_species_id, pfts = pfts, model = model, con = NULL)
+
+  # ### merge with other stuff
   obs$pft <- pft.info$pft
-  
+
   veg_info[[2]] <- obs
   
   #--------------------------------------------------------------------------------------------------#
   # veg2model
-  
   ## Set model-specific functions
-  pkg <- paste0("PEcAn.", model)
+  pkg <- paste0("PEcAn.", model$type)
   do.call("library", list(pkg))
-  fcnx <- paste("veg2model.", model, sep = "")
+  fcnx <- paste("veg2model.", model$type, sep = "")
   if (!exists(fcnx)) {
     PEcAn.logger::logger.severe(paste(fcnx, "does not exist."))
   }else{
     fcn <- match.fun(fcnx)
   }
-  
-  out <- fcn(outfolder, veg_info, start_date, new_site, source)
-  
+# Cohort2Pool -------------------------------------------------------------
+  # read in registration xml for pool specific information
+  register.xml <- system.file(paste0("register.", model$type, ".xml"), package = paste0("PEcAn.", model$type))
+  register     <- PEcAn.data.atmosphere::read.register(register.xml, con = NULL)
```

Comment:
Given that IC is a new addition the register, I'd recommend adding some checks that the register file exists, and also add some checks (and a default) to see if IC tag exists. That way this code will run (though probably with a warning) for existing models (otherwise you need to go and update the registry for every existing model, which honestly you should probably do anyways)